tools/windows-node-installer/pkg/cloudprovider/factory.go

```diff
@@ -0,0 +1,57 @@
+package cloudprovider
+
+import (
+	"fmt"
+
+	v1 "github.com/openshift/api/config/v1"
+	clientset "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/client"
+	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/cloudprovider/aws"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CloudProviderFactory returns a cloud specific interface
+// The factory takes in kubeconfig of an existing OpenShift Cluster and Cloud vendor specific credential file
+// It also requires name of the credential account since there might be more than one accounts in the file
+// Also a directory which should be default to the current working directory where install information
+// such as created instance and security group id should be saved in.
+// The returned cloud specific interface will be able to perform rest of the necessary functions related to creating or
+// destroying an instance
+func CloudProviderFactory(kubeconfigPath, credentialPath, credAccount, dir *string) (Cloud, error) {
+	// Create a new client of the given OpenShift cluster based on kubeconfig
+	kubeclient, err := client.New(kubeconfigPath)
+	if err != nil {
+		return nil, err
+	}
+	// Obtain cluster infrastructure information which includes Name and Platform
+	infra, err := GetInfrastructure(kubeclient)
+	if err != nil {
+		return nil, err
+	}
+
+	// Select platform based on cluster infrastructure information
+	switch platform := infra.Status.PlatformStatus.Type; platform {
+	case v1.AWSPlatformType:
+		return aws.New(kubeclient, infra, credentialPath, credAccount, dir)
+	default:
+		return nil, fmt.Errorf("the '%v' cloud platform is not supported", platform)
+	}
+}
+
+// GetInfrastructure gets information of current Infrastructure referred by the OpenShift client
+// Each client should have only one infrastructure
+// It returns information of the infrastructure including InfrastructureName, cluster region
+func GetInfrastructure(c *clientset.Clientset) (*v1.Infrastructure, error) {
```

Comment:
I'd rather make this a method for the OpenShift cluster. An example would be to create a struct

```
type OpenShift struct {
client *clientset.ClientSet
}

func (*os OpenShift) GetInfraStructure() (*v1.Infrastructure, error) {}
```